index.js

```diff
@@ -94,7 +94,11 @@ module.exports = function typeDetect(obj) {
    * Post:
    *   array literal      x 22,479,650 ops/sec ±0.96% (81 runs sampled)
    */
-  if (isArrayExists && Array.isArray(obj)) {
+  if (
+    isArrayExists &&
+    Array.isArray(obj) &&
+    (symbolExists === false || typeof obj[Symbol.toStringTag] === 'undefined')
+  ) {
```

Comment:
Typically the precedent is that if we have long conditions, they get assigned to a var which is more readable. For example here:

```js
var symbolToStringTagExists = (symbolExists === false || typeof obj[Symbol.toStringTag] === 'undefined');
if (isArrayExists && Array.isArray(obj) && symbolToStringTagExists)
```

Would be ideal. Both lines <120 and the if is more readable as the conditions are given names with semantic meaning.